From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

These patches allow the docs to compile. It is possible that this patch will
not be needed under a different version of haddock.

diff -urN purescript-0.14.1-docs/src/Language/PureScript/TypeChecker/Kinds.hs purescript-0.14.1/src/Language/PureScript/TypeChecker/Kinds.hs
--- purescript-0.14.1-docs/src/Language/PureScript/TypeChecker/Kinds.hs	2021-04-11 12:39:21.000000000 -0600
+++ purescript-0.14.1/src/Language/PureScript/TypeChecker/Kinds.hs	2021-05-13 18:37:10.548926697 -0600
@@ -588,10 +588,8 @@
   )
 
 type DataDeclarationResult =
-  ( [(DataConstructorDeclaration, SourceType)]
-  -- ^ The infered type signatures of data constructors
-  , SourceType
-  -- ^ The inferred kind of the declaration
+  ( [(DataConstructorDeclaration, SourceType)] -- The infered type signatures of data constructors
+  , SourceType                                 -- The inferred kind of the declaration
   )
 
 kindOfData
@@ -639,10 +637,8 @@
   )
 
 type TypeDeclarationResult =
-  ( SourceType
-  -- ^ The elaborated rhs of the declaration
-  , SourceType
-  -- ^ The inferred kind of the declaration
+  ( SourceType -- The elaborated rhs of the declaration
+  , SourceType -- The inferred kind of the declaration
   )
 
 kindOfTypeSynonym
@@ -752,14 +748,10 @@
   )
 
 type ClassDeclarationResult =
-  ( [(Text, SourceType)]
-  -- ^ The kind annotated class arguments
-  , [SourceConstraint]
-  -- ^ The kind annotated superclass constraints
-  , [Declaration]
-  -- ^ The kind annotated declarations
-  , SourceType
-  -- ^ The inferred kind of the declaration
+  ( [(Text, SourceType)] -- The kind annotated class arguments
+  , [SourceConstraint] -- The kind annotated superclass constraints
+  , [Declaration] -- The kind annotated declarations
+  , SourceType -- The inferred kind of the declaration
   )
 
 kindOfClass
diff -urN purescript-0.14.1-docs/src/Language/PureScript/TypeChecker/Roles.hs purescript-0.14.1/src/Language/PureScript/TypeChecker/Roles.hs
--- purescript-0.14.1-docs/src/Language/PureScript/TypeChecker/Roles.hs	2021-04-05 16:01:35.000000000 -0600
+++ purescript-0.14.1/src/Language/PureScript/TypeChecker/Roles.hs	2021-05-13 18:21:28.508687527 -0600
@@ -206,12 +206,12 @@
           -- arguments, recursively infer the roles of the type constructor's
           -- arguments. For each (role, argument) pair:
           --
-          -- * If the role is nominal, mark all free variables in the
+          -- - If the role is nominal, mark all free variables in the
           --   argument as nominal also, since they cannot be coerced if the
           --   argument's nominality is to be preserved.
-          -- * If the role is representational, recurse on the argument, since
+          -- - If the role is representational, recurse on the argument, since
           --   its use of our parameters is important.
-          -- * If the role is phantom, terminate, since the argument's use of
+          -- - If the role is phantom, terminate, since the argument's use of
           --   our parameters is unimportant.
           TypeConstructor _ t1Name ->
             let
